### PR TITLE
Muon: Decouple muon's and its adamw part's weight decay

### PR DESCRIPTION
**Pull request by leloykun**
Decouples muon's and its adamw part's weight decay.

- Learning rates and weight decay are tightly linked and should be sweeped jointly.
- Adam's updates tend to have size `0.23 * adam_lr` RMS while Muon's updates have size `lr` RMS (see Kimi's scaling factor). Different effective update sizes means we should, ideally, use different weight decays for the weights optimized by Muon vs. the weights optimized by AdamW.

### CODE REVIEW

**Review by WhenWen — APPROVED** [review]
Lgtm! We will move on to add a more fine-grained sweeping for this adam weight decay

**Comment by leloykun** [comment]
Thanks @WhenWen ! It does look like I need another approval from you before I could merge this.

**Review by WhenWen — APPROVED** [review]
lgtm. Will test small lm head wd later in Marin
